storage/oasis/consensus.go

```diff
@@ -53,31 +53,40 @@ func (cc *ConsensusClient) GetEpoch(ctx context.Context, height int64) (beaconAP
 }
 
 // AllData returns all relevant data related to the given height.
-func (cc *ConsensusClient) AllData(ctx context.Context, height int64) (*storage.ConsensusAllData, error) {
+func (cc *ConsensusClient) AllData(ctx context.Context, height int64, fastSync bool) (*storage.ConsensusAllData, error) {
 	blockData, err := cc.BlockData(ctx, height)
 	if err != nil {
 		return nil, err
 	}
+
 	beaconData, err := cc.BeaconData(ctx, height)
 	if err != nil {
 		return nil, err
 	}
+
 	registryData, err := cc.RegistryData(ctx, height)
 	if err != nil {
 		return nil, err
 	}
+
 	stakingData, err := cc.StakingData(ctx, height)
 	if err != nil {
 		return nil, err
 	}
-	schedulerData, err := cc.SchedulerData(ctx, height)
-	if err != nil {
-		return nil, err
+
+	var schedulerData *storage.SchedulerData
+	if !fastSync {
```

Comment:
:+1:  Added:
```
// Scheduler data is not needed during fast sync. It contains no events,
	// only a complete snapshot validators/committees. Since we don't store historical data,
	// any single snapshot during slow-sync is sufficient to reconstruct the state.
	```

Reply:
Thanks :+1: